### PR TITLE
docs: add API deprecation details

### LINKED ISSUE #360: Expand deprecation policy to include APIs
Currently [we state a deprecation policy for features](https://github.com/openservicemesh/osm-docs/blob/main/content/docs/releases/feature_lifecycle.md). We should expand this to include APIs so that users are away of supportability.

### PR DESCRIPTION
**Pull request by zr-msft**
…ion details

Signed-off-by: Zach Rhoads <zachary.rhoads@microsoft.com>

Add policy details around how and when APIs may be deprecated or removed.

fixes #360

### CODE REVIEW

**Comment by zr-msft** [comment]
@trstringer let me know what you think

File: content/en/docs/releases/feature_lifecycle.md [hunk 1]
```diff
@@ -104,4 +104,16 @@ foo: bar
...
+A value marked as `DEPRECATED` in a release will only be removed in the next minor release at the earliest. This means, a value deprecated in release `vX.Y.Z` will only be removed on or after release `vX.Y+1.Z`. Deprecated values will be documented in the release notes.
+
+## API Deprecation
+
+Similar to the [Kubernetes API deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/), GA or stable APIs will be marked as deprecated prior to removal.
+
+Stable APIs marked as deprecated:
+
+- Remain functional but will generate a warning if used
+- Will not be removed within a major release version
```
Comment (SanyaKochhar) [inline content/en/docs/releases/feature_lifecycle.md:116]: will these also be removed in the next minor release then?

Comment (zr-msft) [reply content/en/docs/releases/feature_lifecycle.md:116]: I should clarify, the intention was to say the earliest a deprecated feature would be removed is the next major release version. Thank you @SanyaKochhar

Comment (zr-msft) [reply content/en/docs/releases/feature_lifecycle.md]: fixed

**Review by SanyaKochhar — APPROVED** [review]

File: content/en/docs/releases/feature_lifecycle.md (hunk 1 above)
Comment (SanyaKochhar) [reply content/en/docs/releases/feature_lifecycle.md]: great thanks!

**Comment by keithmattix** [comment]
@steeling you probably have some thoughts on this

**Review by keithmattix — APPROVED** [review]
I looked at some of the notes Sean published, and this doesn't conflict with what he's proposing, so I'll go ahead and approve this for now

**Pull request merged**